lib/elixir/test/elixir/kernel/raise_test.exs

```diff
@@ -361,6 +361,34 @@ defmodule Kernel.RaiseTest do
       assert result == "expected a function, got: :example"
     end
 
+    test "badfun error when the function is gone" do
+      defmodule Foo do
+        def fun, do: fn -> :ok end
+      end
+
+      fun = Foo.fun()
+
+      :code.delete(Foo)
+
+      defmodule Foo do
+        def fun, do: fn -> :another end
+      end
+
+      :code.purge(Foo)
+
+      result =
+        try do
+          fun.()
+        rescue
+          x in [BadFunctionError] -> Exception.message(x)
+        end
+
+      regex =
+        ~r/function #Function<[0-9]\.[0-9]*\/0 in Kernel.RaiseTest.Foo> is invalid, likely because it points to an old version of the code/
+
+      assert Regex.match?(regex, result)
```

Comment:
Minor nitpick: 

```elixir
assert result =~ ~r/function #Function<[0-9]\.[0-9]*\/0 in Kernel.RaiseTest.Foo> is invalid, likely because it points to an old version of the code/

# or even 

assert result =~ ~r/function #Function<*\/0 in Kernel.RaiseTest.Foo> is invalid/ 
assert result =~ ~r/likely because it points to an old version of the code/
```